content/ch-gates/oracles.ipynb

```diff
@@ -0,0 +1,289 @@
+    "$$\n",
+    "\n",
+    "For phase oracles, the output $f(x)$ is typically a simple bit value of $0$ or $1$, corresponding to inducing phaes of $+1$ and $-1$, respectively."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Taking Out the Garbage <a id='garbage'></a>\n",
+    "\n",
+    "The functions evaluated by an oracle are typically those that can be evaluated efficiently on a classical computer. However, the need to implement it as a unitary in one of the forms shown above means that it must instead be implemented using quantum gates. However, this is not quite as simple as just taking the Boolean gates that can implement the classical algorithm, and replacing them with their classical counterparts.\n",
+    "\n",
+    "One issue that we must take care of is that of reversibility. A unitary of the form $U = \\sum_x \\left| f(x) \\right\\rangle \\left\\langle x \\right|$ is only possible if every unique input $x$ results in a unique output $f(x)$, which is not true in general. However, we can force it to e true by simply including a copy of the input in the output.\n",
+    "\n",
+    "\n",
+    "However, we can make it reversible by simply adding a copy of the input in the output. Specifically, we wish to achieve the following. It is this that leads us to the form for Boolean oracles as we saw earlier\n",
+    "$$\n",
+    "U_f \\left|x,\\bar 0 \\right\\rangle = \\left| x,f(x) \\right\\rangle\n",
+    "$$\n",
+    "\n",
+    "With the computation written as a unitary, we are able to consider the effect of applying it to superposition states. For example, let us take the superposition over all possible inputs $x$ (unnormalized for simplicity). This will result in a superposition of all possible input/output pairs,\n",
+    "\n",
+    "$$\n",
+    "U_f \\sum_x \\left|x,0\\right\\rangle = \\sum_x \\left|x,f(x)\\right\\rangle.\n",
+    "$$\n",
+    "\n",
+    "When adapting classical algorithms, we also need to take care that these superpositions behave as we need them to. Classical algorithms typically do not only compute the desired output, but will also create additional information along the way. Such additional remnants of a computation do not pose a significant problem classically, and the memory they take up can easily be recovered by deleting them. From a quantum perspective, however, things are not so easy. \n",
+    "\n",
+    "For example, consider the case that a classical algorithm peforms the following process,\n",
+    "$$\n",
+    "V_f \\left|x,\\bar 0, \\bar 0 \\right\\rangle = \\left| x,f(x), g(x) \\right\\rangle\n",
+    "$$\n",
+    "Here we see a third register, which is used as a 'scratchpad' for the classical algorithm. We will refer ro information that is left in this register at the end of the computation is as the 'garbage', $g(x)$. Let us use $V_f$ to denote a unitary that implements the above.\n",
```

Comment:
- We will refer ro information : ro -> to

- in this register at the end of the computation is as the 'garbage', 
   : "is"  and "the" before 'garbage' might not need to be here?